src/hooks/useModal.ts

```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react';
+
+interface ModalProps {
+  showModal: boolean;
+  handleOpenModal: (e: React.MouseEvent) => void;
+  handleCloseModal: () => void;
+}
+
+const useModal = (): ModalProps => {
+  const [showModal, setShowModal] = useState(false);
+
+  const handleOpenModal = (e: React.MouseEvent) => {

```

Comment:
```suggestion
const handleOpenModal = (e: React.MouseEvent<HTMLButtonElement>) => {
    };
```

이벤트 핸들러에 대한 타입을 작성할때 어떤 HTML 요소에 이벤트 핸들러를 걸어주냐에 따라서 `e` (이벤트)가 가질 수 있는 속성이 조금씩 다릅니다. 타입을 명시해줌으로서 그런 부분을 확인 할 수 있어요.